src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,645 @@
+        return;
+    }
+
+    bool isValidUrl = mDelegate.validateUrl(std::string(transportOptions.url.data(), transportOptions.url.size()));
+
+    if (isValidUrl == false)
+    {
+        auto status = static_cast<uint8_t>(StatusCodeEnum::kInvalidURL);
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Invalid Url");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, status);
+        return;
+    }
+
+    if (transportOptions.triggerOptions.triggerType == TransportTriggerTypeEnum::kUnknownEnumValue)
+    {
+        auto status = static_cast<uint8_t>(StatusCodeEnum::kInvalidTriggerType);
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Invalid Trigger type");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, status);
+        return;
+    }
+
+    // Todo: Validate MotionZones list in the TransportTriggerOptionsStruct field
+    // Validate Bandwidth Requirement
+    CHIP_ERROR err = mDelegate.ValidateBandwidthLimit(transportOptions.streamUsage, transportOptions.videoStreamID,
+                                                      transportOptions.audioStreamID);
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Resource Exhausted");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ResourceExhausted);
+        return;
+    }
+
+    // Validate the StreamUsageEnum as per resource management and stream priorities.
+    err =
+        mDelegate.ValidateStreamUsage(transportOptions.streamUsage, transportOptions.videoStreamID, transportOptions.audioStreamID);
+    if (err != CHIP_NO_ERROR)
+    {
+        auto status = static_cast<uint8_t>(StatusCodeEnum::kInvalidStream);
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Invalid Stream");
+        ctx.mCommandHandler.AddClusterSpecificFailure(ctx.mRequestPath, status);
+        return;
+    }
+
+    uint16_t connectionID = GenerateConnectionID();
+
+    if (connectionID == kMaxConnectionId)
+    {
+        ChipLogError(Zcl, "HandleAllocatePushTransport: Max Connections Exhausted");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ResourceExhausted);
+        return;
+    }
+
+    std::shared_ptr<TransportOptionsStorage> transportOptionsPtr = std::make_shared<TransportOptionsStorage>(transportOptions);
+
+    TransportConfigurationStorage outTransportConfiguration(connectionID, transportOptionsPtr);
+
+    Status status = mDelegate.AllocatePushTransport(*transportOptionsPtr, connectionID);
+
+    if (status == Status::Success)
+    {
+        // add connection to CurrentConnections
+        FabricIndex peerFabricIndex = ctx.mCommandHandler.GetAccessingFabricIndex();
+
+        TransportConfigurationStorageWithFabricIndex transportConfiguration({ outTransportConfiguration, peerFabricIndex });
+
+        UpsertStreamTransportConnection(transportConfiguration);
+        response.transportConfiguration = outTransportConfiguration;
```

Comment:
Where is the ExpiryTime field handling from the spec?  Or is the delegate expected to do that?  I don't recall seeing it documented in the delegate API.